addons/website/models/website_menu.py

```diff
@@ -75,6 +101,9 @@ def create(self, vals_list):
         # Only used when creating website_data.xml default menu
         menus = self.env['website.menu']
         for vals in vals_list:
+            if 'parent_id' in vals:
+                self._validate_parent_menu(vals, None)
```

Comment:
I will call ```_validate_parent_menu``` method after calling super of ```create``` and ```write``` so we can easily write logic and iterate over self in ```_validate_parent_menu``` itself.